test/tests/jobs/FailJobTest.cpp

```diff
@@ -0,0 +1,106 @@
+#include <test/lib/BedrockTester.h>
+#include <test/tests/jobs/JobTestHelper.h>
+
+struct FailJobTest : tpunit::TestFixture {
+    FailJobTest()
+        : tpunit::TestFixture("FailJob",
+                              BEFORE_CLASS(FailJobTest::setupClass),
+                              TEST(FailJobTest::nonExistentJob),
+                              TEST(FailJobTest::notInRunningRunqueuedState),
+                              TEST(FailJobTest::failJobInRunningState),
+                              TEST(FailJobTest::failJobInRunqueuedState),
+                              AFTER(FailJobTest::tearDown),
+                              AFTER_CLASS(FailJobTest::tearDownClass)) { }
+
+    BedrockTester* tester;
+
+    void setupClass() { tester = new BedrockTester(_threadID, {{"-plugins", "Jobs,DB"}}, {});}
+
+    // Reset the jobs table
+    void tearDown() {
+        SData command("Query");
+        command["query"] = "DELETE FROM jobs WHERE jobID > 0;";
+        tester->executeWaitVerifyContent(command);
+    }
+
+    void tearDownClass() { delete tester; }
+
+    // Throw an error if the job doesn't exist
+    void nonExistentJob() {
+        SData command("FailJob");
+        command["jobID"] = "1";
+        tester->executeWaitVerifyContent(command, "404 No job with this jobID");
+    }
+
+    // Throw an error if the job is not in RUNNING or REQUEUED state
+    void notInRunningRunqueuedState() {
+        // Create a job
+        SData command("CreateJob");
+        command["name"] = "job";
+        STable response = tester->executeWaitVerifyContentTable(command);
+        string jobID = response["jobID"];
+
+        // Finish it
+        command.clear();
+        command.methodLine = "FailJob";
+        command["jobID"] = jobID;
+        tester->executeWaitVerifyContent(command, "405 Can only fail RUNNING or RUNQUEUED jobs");
+    }
+
+    // Fail job in RUNNING state
+    void failJobInRunningState() {
+        // Create a job
+        SData command("CreateJob");
+        command["name"] = "job";
+        command["retryAfter"] = "+1 MINUTES";
+        STable response = tester->executeWaitVerifyContentTable(command);
+        string jobID = response["jobID"];
+
+        command.clear();
+        command.methodLine = "Query";
+        command["query"] = "UPDATE jobs SET state = 'RUNNING' WHERE jobID = " + jobID + ";";
```

Comment:
This is not a very real test. Instead, run a GetJob to grab this job and make the plugin switch its state to RUNNING. Heh - just like you did below 😄